oak_crypto/src/noise_handshake/client.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2024 Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use alloc::vec::Vec;
+
+pub use crate::noise_handshake::crypto_wrapper::{
+    aes_256_gcm_open_in_place, aes_256_gcm_seal_in_place, ecdsa_verify, hkdf_sha256,
+    p256_scalar_mult, rand_bytes, sha256, sha256_two_part, EcdsaKeyPair, P256Scalar, NONCE_LEN,
+    P256_X962_LEN, SHA256_OUTPUT_LEN, SYMMETRIC_KEY_LEN,
+};
+use crate::noise_handshake::{
+    error::Error,
+    noise::{HandshakeType, Noise},
+    Crypter,
+};
+
+pub struct HandshakeInitiator {
+    noise: Noise,
+    identity_pub_key: Option<[u8; P256_X962_LEN]>,
+    ephemeral_priv_key: P256Scalar,
+}
+
+impl HandshakeInitiator {
+    pub fn new_nk(peer_public_key: &[u8; P256_X962_LEN]) -> Self {
```

Comment:
That's what I implemented originally, and then I refactored it this way.  When passing in the handshake type, I had to also pass in Option<&[u8; P256_X962_LEN]> for the NK use case, and I'll need to add an additional Option<> parameter to support noise-KK.

This way, when I add an additional key to support KK, it will not be a breaking change: existing code will continue to work.  Also, it makes it clear to the user what parameters are required.